tests/Cms/Content/ContentLockTest.php

```diff
@@ -262,4 +261,13 @@ public function testResolveWithRemainingUnlocks()
         $app->impersonate('homer@simpson.com');
         $this->assertTrue($page->lock()->isUnlocked());
     }
+
+    public function testForPageWithNoneExistingDir()
```

Comment:
Shouldn't this test rather be in the test class for `ModelWithContent` where the behavior is implemented?

Reply:
There is no test class for this one since it is an abstract class. And I wouldn't want to put it into all three (user, page, file) or choose one of them.

Reply:
Ah, I see.

There should be a test class for the `ModelWithContent` class though because the methods in it should be tested independently of the features of the dependent classes. I've done that for the `Kirby\Cache\Cache` abstract class for example by creating a [mock class for it](https://github.com/getkirby/kirby/blob/master/tests/Cache/CacheTest.php).